src/overlays/actors/ovl_En_Test6/z_en_test6.c

```diff
@@ -35,50 +53,1278 @@ const ActorInit En_Test6_InitVars = {
+
+        phi_f24 -= this->unk_14C;
+        phi_s2 += this->unk_27E;
+    }
+
+    POLY_OPA_DISP = this->unk_148;
+
+    for (i = 0; i < ARRAY_COUNT(D_80A94910); i++) {
+        if (D_80A94910[i].unk_04 > 0.0f) {
+            D_80A94910[i].unk_14(this, globalCtx, &D_80A94910[i]);
+        }
+    }
+
+    CLOSE_DISPS(globalCtx->state.gfxCtx);
+}
+
+void func_80A9369C(Actor* thisx, GlobalContext* globalCtx2) {
+    EnTest6* this = THIS;
+    GlobalContext* globalCtx = globalCtx2;
+    f32 temp_f20;
+    f32 temp_f22;
+    f32 phi_f26;
+    s16 phi_s2;
+    s16 sp78;
+    s32 i;
+    Player* player = GET_PLAYER(globalCtx);
+    s32 pad;
+
+    if (this) {}
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+
+    this->unk_148 = POLY_OPA_DISP;
+    this->unk_27C += (s16)this->unk_154;
+    this->unk_27E = this->unk_27C * 2;
+    sp78 = (globalCtx->state.frames & 60) * 1024;
+    phi_s2 = this->unk_27C + 0x4000;
+
+    switch (player->transformation) {
+        default:
+            phi_f26 = player->actor.world.pos.y + 40.0f;
+            break;
+
+        case 3:
+            phi_f26 = player->actor.world.pos.y + 40.0f;
+            break;
+
+        case 1:
+            phi_f26 = player->actor.world.pos.y + 40.0f;
+            break;
+
+        case 2:
+            phi_f26 = player->actor.world.pos.y + 40.0f;
+            break;
+
+        case 0:
+            phi_f26 = player->actor.world.pos.y + 40.0f;
+    }
+
+    for (i = 0; i < 51; i++) {
+        temp_f20 = Math_CosS(phi_s2) * this->unk_150;
+        temp_f22 = Math_SinS(phi_s2) * this->unk_150;
+        Matrix_InsertZRotation_s(this->unk_27E, MTXMODE_NEW);
+        Matrix_InsertTranslation(phi_f26, temp_f20, temp_f22, MTXMODE_APPLY);
+        Matrix_Scale(0.85f, 0.85f, 0.85f, MTXMODE_APPLY);
+        Matrix_InsertXRotation_s(phi_s2, MTXMODE_APPLY);
+        Matrix_InsertZRotation_s(sp78, MTXMODE_APPLY);
+
+        gSPMatrix(this->unk_148++, Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        gDPSetPrimColor(this->unk_148++, 0, 0xFF, this->unk_280, this->unk_280, this->unk_280, 255);
+        gDPSetEnvColor(this->unk_148++, 235, 238, 235, 255);
+        gDPSetRenderMode(this->unk_148++, G_RM_FOG_SHADE_A, G_RM_AA_ZB_OPA_SURF2);
+        gSPDisplayList(this->unk_148++, gameplay_keep_DL_07AFB0);
+
+        phi_s2 += 0x505;
+    }
+
+    Lights_PointSetPosition(&this->lights[0].info, player->actor.world.pos.x, player->actor.world.pos.y - 10.0f,
+                            player->actor.world.pos.z);
+    Lights_PointSetColorAndRadius(&this->lights[0].info, 100, 250, 100, 200);
+
+    POLY_OPA_DISP = this->unk_148;
+
+    CLOSE_DISPS(globalCtx->state.gfxCtx);
+}
+
+void func_80A939E8(EnTest6* this, GlobalContext* globalCtx2) {
+    GlobalContext* globalCtx = globalCtx2;
+    Player* player = GET_PLAYER(globalCtx);
+    f32 temp_f20;
+    s32 i;
+
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+
+    switch (this->unk_274) {
+        case 0x5B:
+        case 0x5D:
```

Comment:
This switch case can be in decimal. `unk_274` is used in dec for the rest of the file